backend/src/main/java/com/festago/presentation/GlobalExceptionHandler.java

```diff
@@ -14,27 +22,61 @@
 @RestControllerAdvice
 public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
 
+    private static final String LOG_FORMAT = "\n[🚨ERROR]\n{}: {} ({} {})\n[CALLED BY] {} \n[REQUEST BODY] \n{}";
+    private static final String ERROR_LOG_FORMAT = "\n[🚨ERROR]\n{} ({} {})\n[CALLED BY] {} \n[REQUEST BODY] \n{}";
+    private static final Logger errorLogger = LoggerFactory.getLogger("ErrorLogger");
+
     @ExceptionHandler(BadRequestException.class)
-    public ResponseEntity<ErrorResponse> handle(BadRequestException e) {
-        return ResponseEntity.status(HttpStatus.BAD_REQUEST)
-            .body(ErrorResponse.from(e));
+    public ResponseEntity<ErrorResponse> handle(BadRequestException e, HttpServletRequest request) throws IOException {
+        log(errorLogger::info, e, request);
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(ErrorResponse.from(e));
     }
```

Comment:
다음과 같이 로그를 발생시키면, 로그 레벨이 info가 아닐 때도 로그를 남기려고 불필요한 연산이 필요할 것 같아요!
http://dveamer.github.io/backend/HowToUseSlf4j.html
다음과 같은 게시글 참고해시면 좋을 것 같네요!
만약 가장 쉽게 수정한다면 다음과 같이 수정할 수 있을 것 같아요!
```suggestion
if (errorLogger.isInfoEnabled()) {
    log(errorLogger::info, e, request);
}
```

Reply:
{} 을 활용해 문자열을 포매팅하는 경우에는, 로거가 비활성화되어있을때는 해당 메서드 안 파라미터들은 평가되지 않고, 그에 따라 e.getRequest()와 같은 메서드 호출도 발생하지 않을 것 같네요!

아래는 `logger.info(String format, Object... arguments);`에 적혀있는 주석입니다!
```
Log a message at the INFO level according to the specified format and arguments.
This form avoids superfluous string concatenation when the logger is disabled for the INFO level. However, this variant incurs the hidden (and relatively small) cost of creating an Object[] before invoking the method, even if this logger is disabled for INFO. The variants taking one and two arguments exist solely in order to avoid this hidden cost.
Params:
format – the format string
arguments – a list of 3 or more arguments
```

그에따라 `if (errorLogger.isInfoEnabled())`과 같은 조건문 연산은 하지 않아도 되겠네요
(소중한 자료 감사합니다 글렌🙇🏻‍♀️)

Reply:
네 맞아요
하지만 logger.info(msg, foo.logic()); 같은 로직에서 info 로깅을 하지 않더라도 foo.logic() 메소드는 호출이 발생하게 됩니다.
하지만 foo.logic() 메소드에서 발생하는 비용보다, if문으로 먼저 실행을 하지 않도록 명시적으로 막는 것이 더 효율적이겠죠!